### PR TITLE
Remove annoying `GL error` message from Optifine/Shadersmod

### PR DESCRIPTION
**Pull request by Caedis**
Confirmed working when running the modified Complementary Shaders that Sampsa uses.

### CODE REVIEW

File: src/main/java/com/mitchej123/hodgepodge/asm/transformers/optifine/GLErrorLoggingTransformer.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Override
+    public byte[] transform(String name, String transformedName, byte[] basicClass) {
+        if ("shadersmod.client.Shaders".equals(transformedName)) {
+            LOGGER.info("TRANSFORMING shadersmod.client.Shaders");
+            final ClassReader cr = new ClassReader(basicClass);
+            final ClassNode cn = new ClassNode(ASM5);
+            cr.accept(cn, 0);
+            for (MethodNode m : cn.methods) {
+                if ("checkGLError".equals(m.name)) {
```
Comment (Alexdoru) [inline src/main/java/com/mitchej123/hodgepodge/asm/transformers/optifine/GLErrorLoggingTransformer.java:30]: you should check that the descriptor ends with `)I`

File: src/main/java/com/mitchej123/hodgepodge/asm/transformers/optifine/GLErrorLoggingTransformer.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                    final InsnList i = m.instructions;
+                    i.clear();
+                    i.add(
+                            new MethodInsnNode(
+                                    Opcodes.INVOKESTATIC,
+                                    "org/lwjgl/opengl/GL11",
+                                    "glGetError",
+                                    "()I",
+                                    false));
+                    i.add(new InsnNode(Opcodes.IRETURN));
```
Comment (Alexdoru) [inline src/main/java/com/mitchej123/hodgepodge/asm/transformers/optifine/GLErrorLoggingTransformer.java:41]: you need to change max locals / max stack and probably remove some local variables from the table

**Review by Alexdoru — CHANGES_REQUESTED** [review]

File: src/main/java/com/mitchej123/hodgepodge/asm/transformers/optifine/GLErrorLoggingTransformer.java (hunk 2 above)
Comment (Caedis) [reply src/main/java/com/mitchej123/hodgepodge/asm/transformers/optifine/GLErrorLoggingTransformer.java:41]: I have no idea how. Got any examples?

Comment (Caedis) [reply src/main/java/com/mitchej123/hodgepodge/asm/transformers/optifine/GLErrorLoggingTransformer.java:41]: Is it like this?
`final ClassWriter cw = new ClassWriter(ClassWriter.COMPUTE_MAXS);`

Comment (Alexdoru) [reply src/main/java/com/mitchej123/hodgepodge/asm/transformers/optifine/GLErrorLoggingTransformer.java:41]: that will recompute the MAX, but it won't clean the local variable table

Comment (Caedis) [reply src/main/java/com/mitchej123/hodgepodge/asm/transformers/optifine/GLErrorLoggingTransformer.java:41]: well how can I clean the local variable table? Got an example?

Comment (Alexdoru) [reply src/main/java/com/mitchej123/hodgepodge/asm/transformers/optifine/GLErrorLoggingTransformer.java:41]: this is a field in `MethodNode` you look inside the list and remove the variables that you deleted with your ASM
image

**Review by Alexdoru — CHANGES_REQUESTED** [review]

**Comment by Caedis** [comment]
`maxStack` is still 1 because the value from glGetError is put onto the stack and then return is called.

**Comment by bombcar** [comment]
Spotless Is gonna have a heart attack on that wildcard import

**Review by mitchej123 — APPROVED** [review]

**Pull request merged**